buildpack/telemetry/fluentbit.py

```diff
@@ -0,0 +1,133 @@
+
+
+NAMESPACE = "fluentbit"
+CONF_FILENAME = f"{NAMESPACE}.conf"
+FILTER_FILENAME = "redaction.lua"
+FLUENTBIT_ENV_VARS = {
+    "FLUENTBIT_LOGS_PORT": os.getenv("FLUENTBIT_LOGS_PORT", default="5170"),
+}
+
+
+def _set_default_env(m2ee):
+    for var_name, value in FLUENTBIT_ENV_VARS.items():
+        util.upsert_custom_environment_variable(m2ee, var_name, value)
+
+
+def stage(buildpack_dir, destination_path, cache_path):
+
+    if not is_fluentbit_enabled():
+        return
+
+    util.resolve_dependency(
+        "fluentbit",
+        # destination_path - DOT_LOCAL_LOCATION
+        os.path.join(destination_path, NAMESPACE),
+        buildpack_dir=buildpack_dir,
+        cache_dir=cache_path,
+    )
+
+    for filename in (CONF_FILENAME, FILTER_FILENAME):
+        shutil.copy(
+            os.path.join(buildpack_dir, "etc", NAMESPACE, filename),
+            os.path.join(
+                destination_path,
+                NAMESPACE,
+            ),
+        )
+
+    logging.info("Fluent Bit has been installed successfully.")
+
+
+def update_config(m2ee):
+
+    if not is_fluentbit_enabled():
+        return
+
+    _set_default_env(m2ee)
+
+    util.upsert_logging_config(
+        m2ee,
+        {
+            "type": "tcpjsonlines",
+            "name": "FluentbitSubscriber",
+            "autosubscribe": "INFO",
+            "host": "localhost",
+            "port": FLUENTBIT_ENV_VARS["FLUENTBIT_LOGS_PORT"],
+        },
+    )
+
+
+def run():
+
+    if not is_fluentbit_enabled():
+        return
+
+    fluentbit_dir = os.path.join(
+        os.path.abspath(".local"),
+        NAMESPACE,
+    )
+
+    fluentbit_bin_path = os.path.join(
+        fluentbit_dir,
+        "fluent-bit",
+    )
+
+    fluentbit_config_path = os.path.join(
+        fluentbit_dir,
+        CONF_FILENAME,
+    )
+
+    if not os.path.exists(fluentbit_bin_path):
+        logging.warning(
+            "Fluent Bit is not installed yet. "
+            "Please redeploy your application to complete "
+            "Fluent Bit installation."
+        )
+        return
+
+    logging.info("Starting Fluent Bit...")
+
+    subprocess.Popen(
+        (fluentbit_bin_path, "-c", fluentbit_config_path),
+    )
+
+    # The runtime does not handle a non-open logs endpoint socket
+    # gracefully, so wait until it's up
+    @backoff.on_predicate(backoff.expo, lambda x: x > 0, max_time=10)
+    def _await_logging_endpoint():
+        return socket.socket(socket.AF_INET, socket.SOCK_STREAM).connect_ex(
+            ("localhost", int(FLUENTBIT_ENV_VARS["FLUENTBIT_LOGS_PORT"]))
+        )
+
+    logging.info("Awaiting Fluent Bit log subscriber...")
+    if _await_logging_endpoint() == 0:
+        logging.info("Fluent Bit log subscriber is ready.")
+    else:
+        logging.error(
+            "Fluent Bit log subscriber was not initialized correctly."
+            "Application logs will not be shipped to Fluent Bit."
+        )
+
+    splunk.ready()
```

Comment:
Shouldn't this call be within the `if` block, only when fluent bit log subscriber is ready?

Reply:
moved